lib/BackgroundJobs/SchedulerJob.php

```diff
@@ -24,28 +27,55 @@ class SchedulerJob extends QueuedJob {
 		'OCA\GroupFolders\Mount\MountProvider'
 	];
 
+	public const LOCAL_HOME_MOUNT_TYPE = 'OC\Files\Mount\LocalHomeMountProvider';
+
 	private LoggerInterface $logger;
 	private IDBConnection $db;
 	private IJobList $jobList;
+	/**
+	 * @var \OC\SystemConfig
+	 */
+	private SystemConfig $systemConfig;
 
-	public function __construct(ITimeFactory $timeFactory, Logger $logger, IDBConnection $db, IJobList $jobList) {
+	public function __construct(ITimeFactory $timeFactory, Logger $logger, IDBConnection $db, IJobList $jobList, SystemConfig $systemConfig) {
 		parent::__construct($timeFactory);
 		$this->logger = $logger;
 		$this->db = $db;
 		$this->jobList = $jobList;
+		$this->systemConfig = $systemConfig;
 	}
 
 	protected function run($argument): void {
 		$qb = $this->db->getQueryBuilder();
-		$qb->select('root_id', 'storage_id')
+		$qb->select('root_id', 'storage_id', 'mount_provider_class')
 			->from('mounts')
 			->where($qb->expr()->in('mount_provider_class', $qb->createPositionalParameter(self::ALLOWED_MOUNT_TYPES, IQueryBuilder::PARAM_STR_ARRAY)));
 		$result = $qb->executeQuery();
 
 		while ($row = $result->fetch()) {
+			$storageId = (int)$row['storage_id'];
+			$rootId = (int)$row['root_id'];
+			$overrideRoot = $rootId;
+			if ($row['mount_provider_class'] === self::LOCAL_HOME_MOUNT_TYPE) {
```

Comment:
does this work also with s3 as primary storage or is there different handling ?

Reply:
I'm not sure. How does s3 as primary storage work?